src/mattack_actors.cpp

```diff
@@ -74,6 +78,46 @@ bool leap_actor::call( monster &z ) const
         return false;
     }
 
+    if( !forbidden_effects_any.empty() ) {
```

Comment:
```suggestion
```
That check is unnecessary. The loop will do just fine without it. If `forbidden_effects_any` is empty, it will be skipped anyway.